src/nlp/__init__.py

```diff
@@ -34,3 +35,10 @@
 from .splits import NamedSplit, Split, SplitBase, SplitDict, SplitGenerator, SplitInfo, SubSplitInfo, percent
 from .utils import *
 from .utils.tqdm_utils import disable_progress_bar
+
+
+if int(pyarrow.__version__.split(".")[1]) < 16 or int(pyarrow.__version__.split(".")[0]) > 0:
```

Comment:
This conditional is incorrect. Should be `if int(pyarrow.__version__.split(".")[0]) <= 0 and int(pyarrow.__version__.split(".")[1]) < 16:`